app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt

```diff
@@ -267,6 +274,18 @@ interface ProjectViewModel {
                     .filter({ IntentMapper.appBannerIsSet(it) })
                     .compose(bindToLifecycle())
                     .subscribe { _ -> this.koala.trackOpenedAppBanner() }
+
+            this.isHorizontalRewardsEnabled.onNext(environment.enableHorizontalRewards().get())
```

Comment:
I'm super confused. What's happening here?

Reply:
Here we are just sending the `Output` of the feature flag.

Reply:
I mean for the whole block! Github doesn't do a good job of that. That's just the first line of that code.

Reply:
```this.isHorizontalRewardsEnabled.onNext(environment.enableHorizontalRewards().get())

            val viewId = this.isHorizontalRewardsEnabled
                    .compose(bindToLifecycle())
                    .map { getViewIdToHide(it) }

            Observable.combineLatest(viewId, this.isHorizontalRewardsEnabled)
              { id, enabled -> Pair.create(id, enabled) }
                    .compose(bindToLifecycle())
                    .subscribe(this.viewToHide)
```

This is to get the `feature flag value` then based off of that value retrieve the `id` of the view that we need to hide. Then we create a pair so that in the `activity` we hide the view and create the adapter with the feature flag value, that way we know which ViewHolder to return.